File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = len(nums)
        small = 0
        large = l - 1
        i = 0

        while True:
            i += 1
            index = small + ((large - small) // 2)
            #print(small)
            #print(large)
            #print(nums[index])
            #print()

            if nums[index] == target:
                return index
            elif nums[index] < nums[large]:
                if target > nums[large]:
                    large = index - 1
                elif target < nums[large]:
                    if target < nums[index]:
                        large = index - 1
                    else:
                        small = index + 1
                else:
                    return large
            else:
                if target < nums[small]:
                    small = index + 1
                elif target > nums[small]:
                    if target < nums[index]:
                        large = index - 1
                    else:
                        small = index + 1
                else:
                    return small

            if small > large:
                return -1

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ret;
        ret.reserve(2);

        for (int i = 0; i < nums.size(); i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                int temp = nums[i] + nums[j];

                if (temp == target) {
                    ret.push_back(i);
                    ret.push_back(j);

                    return ret;
                }
            }
        }
    }
};
